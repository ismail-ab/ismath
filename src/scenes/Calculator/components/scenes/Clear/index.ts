export { default as Clear } from './Clear';