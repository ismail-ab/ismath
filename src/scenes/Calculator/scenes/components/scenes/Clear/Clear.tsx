import * as React from 'react';
import { Button } from '@material-ui/core';
import { Delete } from '@material-ui/icons';
import { CalculatorContext } from '../../../Calculator/Calculator';

const Clear: React.SFC = () => (
  <CalculatorContext.Consumer>
    {({ clearCalculator }) => (
      <Button
        variant="fab"
        color="secondary"
        mini
        onClick={() => clearCalculator()}
      >
        <Delete />
      </Button>
    )}
  </CalculatorContext.Consumer>
);

export default Clear;