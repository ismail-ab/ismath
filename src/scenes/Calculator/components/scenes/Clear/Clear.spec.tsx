import { } from 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Clear from './Clear';

describe('Clear', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Clear />);

    expect(wrapper).toMatchSnapshot();
  });
});
