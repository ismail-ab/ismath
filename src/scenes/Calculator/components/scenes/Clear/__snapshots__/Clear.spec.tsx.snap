// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear renders correctly 1`] = `<[object Object] />`;
